compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantValueResolver.java

```diff
@@ -405,7 +412,12 @@ private BLangConstantValue calculateSubtract(BLangConstantValue lhs, BLangConsta
         switch (Types.getReferredType(this.currentConstSymbol.type).tag) {
             case TypeTags.INT:
             case TypeTags.BYTE: // Byte will be a compiler error.
-                result = (Long) lhs.value - (Long) rhs.value;
+                try {
+                    result = Math.subtractExact((Long) lhs.value, (Long) rhs.value);
+                } catch (ArithmeticException ae) {
+                    dlog.error(currentPos, DiagnosticErrorCode.INT_RANGE_OVERFLOW_ERROR);
+                    return new BLangConstantValue(null, this.currentConstSymbol.type);
```

Comment:
We don't have to return here. At this level, the result will be null; if we break in the following line, the last statement will return the null const value. Check other places as well

Reply:
Updated  9c8454389b2969a3679513c74c57a8390d137c18